t/open311.t

```diff
@@ -126,6 +126,28 @@ subtest 'posting service request with basic_description' => sub {
     is $c->param('description'), $problem->detail, 'description correct';
 };
 
+subtest 'posting service request which responds with a token, then fetching id with token' => sub {
+    my $extra = {
+        url => 'http://example.com/report/1',
+    };
+
+    my $results = make_service_req(
+        $problem,
+        $extra,
+        $problem->category,
+        '<?xml version="1.0" encoding="utf-8"?><service_requests><request><token>369FBB0C-4755-11EC-96C0-F6CCC296753D</token></request></service_requests>',
+        { extended_description => 0 },
+    );
+
+    diag $results->{ res };
+
+    is $results->{ res }, 248, 'got request id';
```

Comment:
This won't work because you haven't told the tests what to return for the token response yet.

You'll need to add a `Open311->_inject_response` call to inject a response for the token XML which will contain this request ID.

Reply:
Aye, that's the bit I was working on when I ran out of time.